kernel/kernel-api/src/main/java/io/delta/kernel/internal/rowtracking/MaterializedRowTrackingColumn.java

```diff
@@ -145,6 +148,39 @@ public static Optional<Metadata> assignMaterializedColumnNamesIfNeeded(Metadata
         : Optional.of(metadata.withMergedConfiguration(configsToAdd));
   }
 
+  /**
+   * Converts a logical column to its corresponding physical column if it is a row tracking metadata
+   * column (ROW_ID or ROW_COMMIT_VERSION).
+   *
+   * <p>Note that we must not mark the physical columns as metadata columns because as far as the
+   * parquet reader is concerned, these columns are not metadata columns.
+   *
+   * @param column The logical column to convert.
+   * @param metadata The current metadata of the table.
+   * @return An Optional containing the physical column if conversion is applicable;
+   *     Optional.empty() otherwise.
+   */
+  public static Optional<StructField> convertToPhysicalColumn(
+      StructField column, Metadata metadata) {
+    if (column.equals(METADATA_ROW_ID_COLUMN)) {
```

Comment:
Match metadata column based on their name instead of the whole struct. In the current model, the name is the unique identifier. We don't wnat differences in e.g. nullability to cause metadata columns to be ignored.